### DISCUSSION TITLE
Error w/ FastF1 version?

### DISCUSSION THREAD

**Discussion by SethWalder** [SCasanova/f1dataR#226] [General]
Been awhile since I've used f1dataR and I recently updated the package. However, when I've gone to load telemetry data I get this error

Error in `check_ff1_version()`:
! An old version of FastF1 is in use. f1dataR requires FastF1 version 3.1.0 or newer.
✖ Support for older FastF1 versions was removed in f1dataR v1.6.0
ℹ You can update your FastF1 installation by running: `reticulate::py_install('fastf1')`

I've run the update suggested and best I can tell I have updated FastF1 into version 3.3.0 and in the correct environment but I still get the error. Has anyone else run into this or know the solution?

Appreciate the help (and sorry if this the wrong place to put this question!)

**Comment by SCasanova** [comment]
Hi Seth, 

I also recently had some Python-related issues on my computer (and the interaction with R). 

What's your currently installed Python version and does `f1dataR::get_fastf1_version()` yield any result? 

In my case, I had to manually change the python environment used by reticulate using `Sys.setenv(RETICULATE_PYTHON = ...)` and that solved it but I had a slightly different issue.  (Pick from the resulting list of `reticulate::conda_list()`)

If the problem continues perhaps opening an issue is the best course of action.

**Reply by jvalls99** [reply]
Hi,

When using FastF1 in Python I can access the 2024 season data but in R with f1dataR it states that the most recent season is 2023.

Is this problem only happening to me?

**Reply by SCasanova** [reply]
Thanks for bringing this up. It seems the way it's set up right now only allows for 2024 data to be accessed until after the first race. We will fix this behaviour soon

**Reply by jvalls99** [reply]
Perfect, thanks for the reply. 

And just to let you know that the package is really good, so congratulations and thanks for making F1 data more accessible.

**Reply by SCasanova** [reply]
No problem and thank you for that! We will work to keep it running!

**Comment by pbulsink** [comment]
To add to that, the `reticulate` package which performs the connection to python typically creates or uses its own virtual environment, as to not affect the user's system more widely. When you updated FastF1, did you update the one in the right virtual environment, in contrast to the system one?

As @SCasanova suggested, if you run `get_fastf1_version()`, what's the response?

You should be able to update the correct environment's FastF1 by running `setup_fastf1()` (check the documentstion there if you have a different virtual or Conda environment to pass instead of the default). 

Let us know how this goes, and take a look at the [setup_fastf1 vignette](https://cran.r-project.org/web/packages/f1dataR/vignettes/setup_fastf1.html) too. Let us know if there's anything that could be clearer in this process so we can better our documentation, or make some edits and submit a PR yourself if you'd like!

**Comment by SethWalder** [comment]
Thank you so much for the help @SCasanova and @pbulsink, I got it to work thanks to your help! Don't worry the directions are clear I think I am I just python illiterate!

Seriously grateful to you all for creating/supporting/maintaining/troubleshooting f1dataR.

**Reply by SCasanova** [reply]
Our pleasure, Seth! Do you mind sharing what the problem was in the end? So we can keep track/steer people in the right direction in the future.

**Reply by SethWalder** [reply]
So I think it was that when I was updating fastF1 I was doing it in the wrong environment. 

When I typed in f1dataR::get_fastf1_version() as suggested, I got:

$major
[1] 2

$minor
[1] 3

Once I ran setup_fastf1() and restarted R it worked! Probably would not have been an issue for a first time user but I think just because I'd been using f1dataR awhile ago but not in awhile, it tripped me up? Not totally sure!

**Comment by pbulsink** [comment]
Thanks Seth

@Scasanova maybe we should inform the user in the deprecation warning that
they can update FastF1 manually, or through calling `setup_fastf1()`?

**Comment by SCasanova** [comment]
Yeah, good idea. Make the error more informative
